### PR TITLE
ILP64 reference with helpers 64bit updates

### PR DESCRIPTION
**Pull request by TorreZuk**
* WIP will crash in tests
* Adds OpenMP and AOCL for ILP64 reference

### CODE REVIEW

File: clients/common/norm.cpp [hunk 1]
```diff
@@ -44,122 +46,135 @@
...
+//             int*                  incx,
+//             hipblasDoubleComplex* y,
+//             int*                  incy);
 
 #ifdef __cplusplus
 }
 #endif
 
+template <typename T>
+void m_axpy_64(int64_t N, T* alpha, T* x, int64_t incx, T* y, int64_t incy)
```
Comment (daineAMD) [inline clients/common/norm.cpp:76]: Why do we need this one here?

File: clients/include/hipblas_vector.hpp [hunk 2]
```diff
@@ -516,13 +533,13 @@ void hipblas_init_alternating_template(U& that, T rand_gen(), T rand_gen_alt(),
 
     for(int b = 0; b < that.batch_count(); ++b)
```
Comment (daineAMD) [inline clients/include/hipblas_vector.hpp]: should be int64_t here

File: clients/common/norm.cpp (hunk 1 above)
Comment (TorreZuk) [reply clients/common/norm.cpp:76]: What do you mean exactly, why not reuse ref_  form?   It alllows a self contained 64bit support in norm.cpp.  Windows will get additions to allow 32bit but I think linux may be stuck with always 64bit reference.   Due to variations it maybe only solver -> LAPACKE will be wrapped with #ifdef WIN32 to allow 32bit API visibility.

**Review by daineAMD — APPROVED** [review]
Cuda pipeline can't find omp looks like.

**Comment by TorreZuk** [comment]
> Cuda pipeline can't find omp looks like.

Okay thanks, I'll probably have to experiment with the docker.  No shortage of problems to work around.

**Comment by TorreZuk** [comment]
Okay @daineAMD if you can re-review as a lot has changed but this is working on solver now with libflame for lapack.  So is windows.  Unfortunately seems like cuda pipline is stopped were you already inquiring on this or should I ?

**Review by daineAMD — APPROVED** [review]
Great to have the matrix init code, LGTM I think.

**Pull request merged**